show/main.py

```diff
@@ -1804,6 +1804,716 @@ def tablelize(keys, data, enable_table_keys, prefix):
+                    checkin = 1
+
+    if (checkin == 1):
+        #health pre = (P/E cycle - AVG erese)/ P/E cycle
+        healthpre = (cycle - avgerase) * 100.0 / cycle
+        click.echo("Device Health Status Is: {0:.2f}%".format(healthpre))
+    else:
+        click.echo("Can't get Average_Erase_Count attributes")
+
+    echo_empty_line()
+
+
+# 
+# 'ssd_badblock' command ("show ssd_badblock /dev/xxx")
+# use this command to Check the later bad block status 
+#
+
+@ssd.command()
+@click.argument("device")
+def badblock(device):
+    """show ssd bad block status which may create"""
+
+    checkin = 0
+    testname = "SSD Badblock Test"
+    command = "sudo smartctl -A " + device
+    proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+    output = proc.stdout.readlines()
+    (out, err) = proc.communicate()
+
+    print_test_title(testname)
+    #smartctl cmd return failed
+    if proc.returncode > 0:
+        for line in output:
+            click.echo(line.strip())
+        return
+    else:
+        for line in output:
+            if "Later_Bad_Block" in line:
+                rawval = line.split()[-1]
+                click.echo("Later_Bad_Block:         {0}".format(rawval))
+                checkin += 1
+
+            if "Later_Bad_Blk_Inf_R/W/E" in line:
+                rawval_read = line.split()[-3]
+                click.echo("Later_Bad_Blk_Inf Read:  {0}".format(rawval))
+                rawval_write = line.split()[-2]
+                click.echo("Later_Bad_Blk_Inf Write: {0}".format(rawval))
+                rawval_erase = line.split()[-1]
+                click.echo("Later_Bad_Blk_Inf Erase: {0}".format(rawval))
+                checkin += 1
+
+    if(checkin != 2):
+        click.echo("Can't get all 'Later_Bad_Block' attributes")
+
+    echo_empty_line()
+
+
+# 
+# 'ssd_temperature' command ("show ssd temperature /dev/xxx")
+# use this command to show SSD temperature
+#
+
+@ssd.command()
+@click.argument("device")
+def temperature(device):
+    """show ssd temperature"""
+
+    checkin = 0
+    testname = "SSD Temperature Test"
+    command = "sudo smartctl -A " + device
+    proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+    output = proc.stdout.readlines()
+    (out, err) = proc.communicate()
+
+    print_test_title(testname)
+    #smartctl cmd return failed
+    if proc.returncode > 0:
+        for line in output:
+            click.echo(line.strip())
+        return
+    else:
+        for line in output:
+            if "Temperature_Celsius" in line:
+                rawval = line.split()[9]
+                temperature = int(rawval)
+                checkin = 1
+
+    if(checkin == 1):
+        click.echo("Temperature_Celsius: {0}C".format(temperature))
+    else:
+        click.echo("Can't get 'Temperature_Celsius' attributes")
+
+    echo_empty_line()
+
+
+#
+# 'ssd all' command 
+# execute all test iterms of SSD
+#
+
+@ssd.command()
+@click.argument("device")
+def all(device):
```

Comment:
all() can call other functions and not duplicate the code...

Reply:
OK,I'll update it as your suggestion.Thanks